virtcontainers/kata_agent.go

```diff
@@ -175,8 +175,20 @@ func (k *kataAgent) generateVMSocket(id string, c KataAgentConfig) error {
 	if c.UseVSock {
 		// We want to go through VSOCK. The VM VSOCK endpoint will be our gRPC.
 		k.Logger().Debug("agent: Using vsock VM socket endpoint")
-		// We dont know yet the context ID - set empty vsock configuration
-		k.vmSocket = kataVSOCK{}
+
+		vhostFd, contextID, err := utils.FindContextID()
+		if err != nil {
+			return err
+		}
+
+		if k.vmSocket == nil {
```

Comment:
virty-wizards, quick Q here. Should we ignore `generateVMSocket` if it ever gets called twice or do we get a new contextID everytime this function is invoked?